store/v2/root/factory.go

```diff
@@ -32,16 +34,20 @@ const (
 	SCTypeIavlV2 SCType = 1
 )
 
+type Options struct {
+	SSType          SSType               `mapstructure:"ss-type" toml:"ss-type"`
+	SCType          SCType               `mapstructure:"sc-type" toml:"sc-type"`
+	SSPruningOption *store.PruningOption `mapstructure:"ss-pruning-option" toml:"ss-pruning-option"`
+	SCPruningOption *store.PruningOption `mapstructure:"sc-pruning-option" toml:"sc-pruning-option"`
+	IavlConfig      *iavl.Config         `mapstructure:"iavl-config" toml:"iavl-config"`
+}
+
 type FactoryOptions struct {
-	Logger          log.Logger
-	RootDir         string
-	SSType          SSType
-	SCType          SCType
-	SSPruningOption *store.PruningOption
-	SCPruningOption *store.PruningOption
-	IavlConfig      *iavl.Config
-	StoreKeys       []string
-	SCRawDB         corestore.KVStoreWithBatch
+	Logger    log.Logger
+	RootDir   string
+	Options   *viper.Viper
```

Comment:
Can we just have a struct here and do the correct filling in runtime?
We cannot expect someone to build `FactoryOptions` and know what the viper should contain.
Additionally it avoids adding viper as a dep here.